pkg/volume/util/nestedpendingoperations/nestedpendingoperations.go

```diff
@@ -58,7 +62,8 @@ type NestedPendingOperations interface {
 	// concatenation of volumeName and podName is removed from the list of
 	// executing operations allowing a new operation to be started with the
 	// volumeName without error.
-	Run(volumeName v1.UniqueVolumeName, podName types.UniquePodName, generatedOperations types.GeneratedOperations) error
+	// TODO (verult) update comment
+	Run(volumeName v1.UniqueVolumeName, podName volumetypes.UniquePodName, nodeName types.NodeName, generatedOperations volumetypes.GeneratedOperations) error
```

Comment:
Are we ever going to have all three specified at once (volumeName, podName, and nodeName)? If not, then maybe we should just support two keys, and make the names generic?

Reply:
I think attach_detach controller only needs volume and node name, kubelet volume manager needs volume and pod name. Then name the second parameter podOrNodeName?

Reply:
Or make it even more generic (not storage specific): `key1` and `key2` or something

Reply:
Ideally I'd like to have a single key param with a generic `OperationKey` type that doesn't even have to be a string (with the `operationKey` struct defined in this PR as an example`. Might be a pain to refactor but I'll give it a try!